src/Microsoft.SqlTools.ServiceLayer/HostLoader.cs

```diff
@@ -0,0 +1,96 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Credentials;
+using Microsoft.SqlTools.ServiceLayer.Extensibility;
+using Microsoft.SqlTools.ServiceLayer.Hosting;
+using Microsoft.SqlTools.ServiceLayer.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.LanguageServices;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.SqlContext;
+using Microsoft.SqlTools.ServiceLayer.Workspace;
+
+namespace Microsoft.SqlTools.ServiceLayer
+{
+    /// <summary>
+    /// Provides support for starting up a service host. This is a common responsibility
+    /// for both the main service program and test driver that interacts with it
+    /// </summary>
+    public static class HostLoader
+    {
+
+        internal static ServiceHost CreateAndStartServiceHost(SqlToolsContext sqlToolsContext)
+        {
+
+            // Grab the instance of the service host
+            ServiceHost serviceHost = ServiceHost.Instance;
+            serviceHost.Initialize();
+
+            InitializeRequestHandlersAndServices(serviceHost, sqlToolsContext);
+
+            // Start the service only after all request handlers are setup. This is vital
+            // as otherwise the 
+            serviceHost.Start().Wait();
+            return serviceHost;
+        }
+
+        internal static void InitializeRequestHandlersAndServices(ServiceHost serviceHost, SqlToolsContext sqlToolsContext)
```

Comment:
>InitializeRequestHandlersAndServices [](start = 29, length = 36)

InitializeRequestHandlersAndServices and InitializeHostServices should be private